apps/scan/frontend/src/app.test.tsx

```diff
@@ -153,11 +160,14 @@ test('app can load and configure from a usb stick', async () => {
 
   // Insert a USB with a ballot package
   apiMock.expectGetUsbDriveStatus('mounted');
+  const { promise: configurePromise, resolve: configureResolve } =
+    deferred<Result<void, BallotPackageConfigurationError>>();
   apiMock.mockApiClient.configureFromBallotPackageOnUsbDrive
     .expectCallWith()
-    .resolves(ok());
+    .returns(configurePromise);
```

Comment:
If this ends up being a common pattern we might build it into the API mock. E.g. `.resolves(value, { deferred: true })` could return a `deferred`

Reply:
That could be neat! It's a bit unintuitive to switch from `resolves(value)` to `returns(promise)`, so a `resolves` API would be easier to use.